### PR TITLE
Support for Kubernetes v1.24

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area open-source usability
/kind enhancement
/platform aws

**What this PR does / why we need it**:
This PR adds support for Kubernetes 1.24 to the extension.

**Which issue(s) this PR fixes**:
Part of gardener/gardener#5912

**Special notes for your reviewer**:

I have successfully validated the functionality as follows:
- ✅ Create new clusters with versions < 1.24
- ✅ Create new clusters with version = 1.24
- ✅ Upgrade old clusters from version 1.23 to version 1.24
- ✅ Delete clusters with versions < 1.24
- ✅ Delete clusters with version = 1.24

**Release note**:

```feature user
The AWS extension does now support shoot clusters with Kubernetes version 1.24. You should consider the [Kubernetes release notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md) before upgrading to 1.24. 
```
```other operator
The following image is updated:
- eu.gcr.io/gardener-project/kubernetes/cloud-provider-aws: 1.23.6 -> 1.24.0(For shoots with Kubernetes version 1.24)
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ This extension controller supports the following Kubernetes versions:
 
 | Version         | Support     | Conformance test results |
 | --------------- | ----------- | ------------------------ |
+| Kubernetes 1.23 | 1.24.0+     | N/A |
```
Comment (rfranzke) [inline README.md:26]:
```suggestion
| Kubernetes 1.24 | 1.24.0+     | N/A |
```

**Review by rfranzke — COMMENTED** [review]
Please also incorporate the release notes of https://github.com/gardener/cloud-provider-aws/releases/tag/v1.24.0

**Comment by acumino** [comment]
> /hold until #550 is merged

Can close #550 as this PR already has changes of #550.

**Comment by rfranzke** [comment]
I would prefer to keep it separately so that we can easier track down the changes. It's approved and merged now, you can drop these commits and rebase on top of `master`.

**Review by rfranzke — APPROVED** [review]
Well done!

/lgtm
/unhold

**Pull request merged**